src/internal/observe/observe.go

```diff
@@ -199,7 +199,10 @@ func MessageWithCompletion(
 			select {
 			case <-contxt.Done():
 				bar.SetTotal(-1, true)
-			case <-ci:
+			case _, ok := <-ci:
+				if !ok {
```

Comment:
This should probably move to *after* we call `SetTotal`. 

Also need to handle this correctly for when the context is cancelled